packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/OutputTypes/OutputLibrary.cs

```diff
@@ -1,45 +1,68 @@
+using System;
 using System.Collections.Generic;
+using Microsoft.Generator.CSharp.Input;
 
 namespace Microsoft.Generator.CSharp
 {
     public class OutputLibrary
     {
+        private IReadOnlyList<EnumTypeProvider>? _enums;
         private IReadOnlyList<ModelTypeProvider>? _models;
         private IReadOnlyList<ClientTypeProvider>? _clients;
 
         public OutputLibrary()
         {
-            EnumMappings = new Dictionary<InputEnumType, EnumType>();
+            EnumMappings = new Dictionary<InputEnumType, EnumTypeProvider>();
             ModelMappings = new Dictionary<InputModelType, ModelTypeProvider>();
-        }
 
-        public IReadOnlyList<ModelTypeProvider> Models => _models ??= BuildModels();
-        public IReadOnlyList<ClientTypeProvider> Clients => _clients ??= BuildClients();
+            _allModels = new(InitializeAllModels);
+        }
 
-        public IDictionary<InputEnumType, EnumType> EnumMappings { get; }
-        public IDictionary<InputModelType, ModelTypeProvider> ModelMappings { get; }
+        private readonly Lazy<(EnumTypeProvider[] Enums, ModelTypeProvider[] Models)> _allModels;
 
-        public virtual ModelTypeProvider[] BuildModels()
+        private (EnumTypeProvider[] Enums, ModelTypeProvider[] Models) InitializeAllModels()
         {
             var input = CodeModelPlugin.Instance.InputLibrary.InputNamespace;
 
-            var modelsCount = input.Models.Count;
-            ModelTypeProvider[] modelProviders = new ModelTypeProvider[modelsCount];
-
-            for (int i = 0; i < modelsCount; i++)
+            var enums = new EnumTypeProvider[input.Enums.Count];
+            for (int i = 0; i < enums.Length; i++)
             {
-                var model = input.Models[i];
-                var typeProvider = new ModelTypeProvider(model, null);
+                var inputEnum = input.Enums[i];
+                var enumType = new EnumTypeProvider(inputEnum, null);
+                enums[i] = enumType;
+                EnumMappings.Add(inputEnum, enumType);
+            }
 
-                modelProviders[i] = typeProvider;
-                ModelMappings.Add(model, typeProvider);
+            var models = new ModelTypeProvider[input.Models.Count];
+            for (int i = 0; i < models.Length; i++)
+            {
+                var inputModel = input.Models[i];
+                var model = new ModelTypeProvider(inputModel, null);
+                models[i] = model;
+                ModelMappings.Add(inputModel, model);
             }
 
-            return modelProviders;
+            return (enums, models);
+        }
+
+        public IReadOnlyList<EnumTypeProvider> Enums => _enums ??= BuildEnums();
+        public IReadOnlyList<ModelTypeProvider> Models => _models ??= BuildModels();
+        public IReadOnlyList<ClientTypeProvider> Clients => _clients ??= BuildClients();
+
+        public IDictionary<InputEnumType, EnumTypeProvider> EnumMappings { get; }
```

Comment:
I don't think we want these here they should be in the typefactory since that is what converts input -> output but I think this can be taken care of as part of https://github.com/microsoft/typespec/issues/3360

Reply:
yeah this is also not only for enums, for models as well therefore we should make another PR to resolve this.